### PR TITLE
Add litellm wrapper

### PR DESCRIPTION
**Pull request by garyzhang99**
## Description

Add litellm chat wrapper to support more models

Now only implement chats.

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: docs/sphinx_doc/en/source/tutorial/203-model.md [hunk 1]
```diff
@@ -87,6 +88,7 @@ In the current AgentScope, the supported `model_type` types, the corresponding
 | ollama                 | Chat            | [`OllamaChatWrapper`](https://github.com/modelscope/agentscope/blob/main/src/agentscope/models/ollama_model.py)                 | `"ollama_chat"`               | llama2, ...                                      |
 |                        | Embedding       | [`OllamaEmbeddingWrapper`](https://github.com/modelscope/agentscope/blob/main/src/agentscope/models/ollama_model.py)            | `"ollama_embedding"`          | llama2, ...                                      |
 |                        | Generation      | [`OllamaGenerationWrapper`](https://github.com/modelscope/agentscope/blob/main/src/agentscope/models/ollama_model.py)           | `"ollama_generate"`           | llama2, ...                                      |
+| LiteLLM API | Chat               | [`LiteLLMModelWrapperChat`](https://github.com/modelscope/agentscope/blob/main/src/agentscope/models/litellm_model.py)             | `"litellm_chat"`                  | -                                                |
```
Comment (qbc2016) [inline docs/sphinx_doc/en/source/tutorial/203-model.md:91]: 'LiteLLMModelWrapperChat' -> 'LiteLLMChatWrapper', same for the below

Comment (garyzhang99) [reply docs/sphinx_doc/en/source/tutorial/203-model.md]: Done

File: src/agentscope/models/litellm_model.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            else:
+                # Merge all messages into a dialogue history prompt
+                dialogue.append(
+                    f"{unit.name}: {_convert_to_str(unit.content)}",
+                )
+
+        dialogue_history = "\n".join(dialogue)
+
+        if sys_prompt is None:
+            user_content_template = "## Dialogue History\n{dialogue_history}"
```
Comment (DavdGao) [inline src/agentscope/models/litellm_model.py:218]: What if dialogue_history is empty?

File: src/agentscope/models/litellm_model.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+        if litellm is None:
+            raise ImportError(
+                "Cannot find litellm package in current python environment.",
+            )
+
+        self.model_name = model_name
+        self.generate_args = generate_args or {}
+        self.api_key = api_key
+        self.api_key_name = api_key_name
```
Comment (DavdGao) [inline src/agentscope/models/litellm_model.py:61]: One questions is that should we provide some hints for this api_key_name? 
Or we remove api_key and api_key_name arguments, and default that all api_key are obtained from environment.

**Review by DavdGao — COMMENTED** [review]
plz see inline comments

File: src/agentscope/service/service_toolkit.py [hunk 4]
```diff
@@ -492,7 +492,8 @@ def bing_search(query: str, api_key: str, num_results: int=10):
         )
 
         # The arguments that requires the agent to specify
-        args_agent = set(argsspec.args) - set(kwargs.keys())
+        # to support class method, the self args are deprecated
```
Comment (DavdGao) [inline src/agentscope/service/service_toolkit.py:495]: Leave the modification of service toolkit for class-level service in the other PR. Don't mix it in this litellm PR.

File: src/agentscope/service/service_toolkit.py [hunk 5]
```diff
@@ -649,7 +650,8 @@ def bing_search(query: str, api_key: str, num_results: int=10):
         )
 
         # The arguments that requires the agent to specify
-        args_agent = set(argsspec.args) - set(kwargs.keys())
+        # we remove the self argument, for class methods
```
Comment (DavdGao) [inline src/agentscope/service/service_toolkit.py:653]: The same as above

File: src/agentscope/models/litellm_model.py [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+        """
+
+        if model_name is None:
+            model_name = config_name
+            logger.warning("model_name is not set, use config_name instead.")
+
+        super().__init__(config_name=config_name)
+
+        if litellm is None:
+            logger.error(
```
Comment (DavdGao) [inline src/agentscope/models/litellm_model.py:61]: Duplicated error report. Suggested to remove `logger.error("xxx")`

**Review by DavdGao — COMMENTED** [review]
Please see inline comments

File: src/agentscope/service/service_toolkit.py (hunk 4 above)
Comment (garyzhang99) [reply src/agentscope/service/service_toolkit.py]: Removed

**Review by DavdGao — APPROVED** [review]
LGTM

**Pull request merged**